### PR TITLE
Add study activity calendar

### PR DESCRIPTION
**Pull request by derricklee43**
https://user-images.githubusercontent.com/39753553/203847830-477e6810-f0da-453c-ade9-46711d420117.mov

### CODE REVIEW

File: src/components/study-activity-calendar/study-activity-calendar.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+export const StudyActivityCalendar = ({ className = '' }: StudyActivityCalendarProps) => {
+  const [currentYear, setCurrentYear] = useState(new Date().getFullYear());
+  const [annualDailyScores, setAnnualDailyScores] = useState<AnnualDailyScores>({});
+  const { getAnnualDailyStudyRecords } = useStudyActivity();
+
+  // fetch annual daily scores, first on load then on current year change
+  useEffect(() => {
+    fetchAnnualStudyRecordsAndRefresh(currentYear);
+  }, [currentYear, setAnnualDailyScores]);
```
Comment (jiejasonliu) [inline src/components/study-activity-calendar/study-activity-calendar.tsx:21]: is `setAnnualDailyScores` in dep needed?

File: src/components/calendar-graph/calendar-graph.scss [hunk 2]
```diff
@@ -0,0 +1,155 @@
+@import '@/assets/colors';
+@import '@/assets/styles';
+
+$month-label-hight: 20px;
+$vertical-margin: 10px;
+
+@mixin hidable-label {
```
Comment (jiejasonliu) [inline src/components/calendar-graph/calendar-graph.scss:7]: spelling if you care at all XD

**Review by jiejasonliu — APPROVED** [review]
this is clean af!!! lgtm

File: src/components/study-activity-calendar/study-activity-calendar.tsx (hunk 1 above)
Comment (derricklee43) [reply src/components/study-activity-calendar/study-activity-calendar.tsx]: Lol, nope. It's fixed now

File: src/components/calendar-graph/calendar-graph.scss (hunk 2 above)
Comment (derricklee43) [reply src/components/calendar-graph/calendar-graph.scss]: my spell checker did me dirty:sob::sob:. It said `hidable` was good but `hideable` wasn't recognized.

File: src/components/calendar-graph/calendar-graph.tsx [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    const monthPairs = [
+      ['jan', 'feb'],
+      ['mar', 'apr'],
+      ['may', 'jun'],
+      ['jul', 'aug'],
+      ['sep', 'oct'],
+      ['nov', 'dec'],
+    ];
+
+    const monthPairLabels = monthPairs.map(([m1, m2]) => (
```
[the remark this answers is not in the document]
Comment (derricklee43) [reply src/components/calendar-graph/calendar-graph.tsx:117]: To make the calendar graph more responsive, when it reaches a certain size, half of the month labels disappear. I'm using this pair structure to do it 
Screen Shot 2022-11-26 at 8 15 53 PM

[the remark this answers is not in the document]
Comment (jiejasonliu) [reply src/components/calendar-graph/calendar-graph.tsx:117]: that makes sense, i like that!

File: src/components/calendar-graph/calendar-graph.scss (hunk 2 above)
Comment (jiejasonliu) [reply src/components/calendar-graph/calendar-graph.scss]: webster dictionary has me questioning what is right... but for some reason `hidable` looks so wrong 😆

**Pull request merged**